### ISSUE TITLE
Image Editor: Enhancement: when making no changes and clicking done - don't save new image

### ISSUE THREAD

**Issue by alisterscott** [Automattic/wp-calypso#7989]
When I make zero changes, but click “Done” instead of “Cancel”, an exact copy is made in my library. Would it be possible to detect zero changes made and not create the copy (since the original image is the same)?

**Comment by ryanboren** [comment]
Reproduced in Firefox, macOS.

The copy on Done flow is awkward. 

The flow:

- Click + to open media modal
- Hover over the desired image
- Click the pencil icon
- Click Edit Image
- Make changes
- Click Done
- Bounced back to the media library - skipping past the image meta modal - where a new edited image awaits
- The new image does not copy the meta fields of the original image

This flow is awkward when wanting to edit an image and its meta at the same time or when wanting to edit the image in place. Consider this: you click the pencil icon, make changes to meta fields, edit the image, click Done, and get popped back to the media lib with a new image containing none of your meta edits. Confusing #forked-flow.

**Comment by rralian** [comment]
> Click the pencil icon

I didn't even realize we offered a pencil icon until I adjusted the thumbnail sizes. The more you know.

Good suggestion. So the ideal flow upon finishing image-editing is to take you back to the image-details modal. I agree. It should also copy the relevant meta details (caption, alt text, description). I'm going to turn those into separate issues (#8074 and #8075) and let this issue stand on its own (which is being addressed by #8033).
